compiler/luci/pass/src/ReplaceSubWithAddPass.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/ReplaceSubWithAddPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+namespace
+{
+
+bool replace_sub_with_const_rhs(luci::CircleSub *sub)
+{
+  auto const_rhs = dynamic_cast<luci::CircleConst *>(sub->y());
```

Comment:
Only y position is supported?

Reply:
For now, yes :)
Only `CircleSub(x, y) whose y is constant` supported

Reply:
X - Y = X + (-Y), where Y is Const
but when X is Const,  X - Y != (-X) + Y
Am I right?

Reply:
It really is :)

Reply:
Hmm... the equation `X - Y != (-X) + Y` is always correct regardless of "X is const?" so the answer is yes...